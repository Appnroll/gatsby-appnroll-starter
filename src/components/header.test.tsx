import { render } from "@testing-library/react"
import Enzyme, { shallow } from "enzyme"
import React from "react"
import Adapter from "enzyme-adapter-react-16"
import styled from "styled-components"
import Header from "./header"

Enzyme.configure({ adapter: new Adapter() })

interface Props {
  text: string
}

const Paragraph = styled.p`
  color: red;
`

const TestHeader: React.FC<Props> = ({ text }) => (
  <div data-testid="hello">
    <Paragraph>{text}</Paragraph>
  </div>
)

describe(`Header`, () => {
  it(`renders Header`, () => {
    const { getByTestId } = render(<TestHeader text="Hello, World!" />)
    expect(getByTestId("hello").textContent).toBe("Hello, World!")
  })
})

describe(`Header`, () => {
  it(`renders Header`, () => {
    const wrapper = shallow(<Header siteTitle="title" />)
    const text = wrapper.find("h1").text()
    expect(text).toEqual("title")
  })
})

