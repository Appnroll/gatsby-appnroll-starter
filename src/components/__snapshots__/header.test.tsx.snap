// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders Header 1`] = `ShallowWrapper {}`;
